pkg/workloadmeta/collectors/internal/containerd/container_builder.go

```diff
@@ -100,3 +111,40 @@ func extractStatus(status containerd.ProcessStatus) workloadmeta.ContainerStatus
 
 	return workloadmeta.ContainerStatusUnknown
 }
+
+// extractIP gets the IP of a container.
+//
+// The containerd client does not expose the IPs, that's why we use the helpers
+// that we have in the "system" package to extract that information from
+// "/proc".
+//
+// A current limitation is that if a container exposes multiple IPs, this
+// function just returns one of them. That means that if a container is attached
+// to multiple networks this might not work as expected.
+func extractIP(container containerd.Container, containerdClient cutil.ContainerdItf) (string, error) {
+	taskPids, err := containerdClient.TaskPids(container)
+	if err != nil {
+		return "", err
+	}
+
+	if len(taskPids) == 0 {
+		return "", errors.New("no PIDs found")
+	}
+
+	IPs, err := system.ParseProcessIPs(
+		config.Datadog.GetString("container_proc_root"),
+		int(taskPids[0].Pid), // Any PID of the container should work
+	)
+	if err != nil {
+		return "", err
+	}
+
+	// From all the IPs, just return the first one that's not localhost.
```

Comment:
Can be added to the `filterFunc` if you implement one :)

Reply:
Added in this commit: https://github.com/DataDog/datadog-agent/pull/11924/commits/9680a8038f6f89a55fe4a7499908dbb87f34d0e8

There are some things that are still hardcoded in the function to which the filter is not applied, but I prefer to leave it like this for now for simplicity. We can revisit this is we add more callers to this func in the future.